client/changelist/files_changelist.go

```diff
@@ -0,0 +1,103 @@
+package changelist
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"sort"
+	"time"
+
+	"code.google.com/p/go-uuid/uuid"
+)
+
+type fileChangelist struct {
+	dir string
+}
+
+func NewFileChangelist(dir string) (*fileChangelist, error) {
+	err := os.MkdirAll(dir, 0600)
+	if err != nil {
+		return nil, err
+	}
+	return &fileChangelist{dir: dir}, nil
+}
+
+func (cl fileChangelist) List() []Change {
+	changes := make([]Change, 0)
+	dir, err := os.Open(cl.dir)
+	if err != nil {
+		return changes
```

Comment:
All the directory paths, both for reading and writing, are generated from a combination of user input and static configuration. If a user is messing with the configuration they're going to get weird behaviour. 

Assuming user's aren't messing with configuration (which they shouldn't unless they really know what they're doing), a non-existent directory indicates no changes are pending, which is functionally identical to an empty directory.